File: Practice/17/Python/casino.py
def get_numbers():
    return 37

def get_k():
    return 12

count = {}
round_appearance = {}
for i in range(get_numbers()):
    count[i] = round_appearance[i] = 0

highest_number = round_now = reds = blacks = 0
while (True):
    round_now += 1
    inpt = int(input())
    if inpt < 0:
        break

    count[inpt] += 1
    round_appearance[inpt] = round_now
    
    highest_number = max(count.values())
    for i in range(get_numbers()):
        if count[i] == highest_number:
            print(i, end = ' ')
    print()

    for i in range(get_numbers()):
        if round_appearance[i] == 0 or round_now-round_appearance[i]>get_k():
            print(i, end = ' ')
    print()

    if inpt != 0:
        if inpt <= 10 or (inpt > 18 and inpt <= 28):
            if inpt % 2 == 1: reds += 1
            else:             blacks += 1
        else:
            if inpt % 2 == 0: reds += 1
            else:             blacks += 1

    print(reds,blacks,'\n')

File: Practice/17/C++/casino.cpp
#include <iostream>

const int NUMBERS = 37;
const int k = 12;

int main() {
    int count[NUMBERS];            // Сколько раз выпало
    int round_appearance[NUMBERS]; // Последний раунд, когда число выпало
    for (int i = 0; i < NUMBERS; i++)
        count[i] = round_appearance[i] = 0;
    
    int input;
    int round_now = 0;
    int blacks = 0, reds = 0;
    while (++round_now) {
        std::cin >> input;
    	if (input < 0)
    	    break;
    	    
        count[input]++;
        round_appearance[input] = round_now;
            
        int highest_number = -1;
    	for (int i = 0; i < NUMBERS; i++)
    	    if (count[i] > highest_number)
    	        highest_number = count[i];
    	
    	for (int i = 0; i < NUMBERS; i++)
            if (count[i] == highest_number)
                std::cout << i << " ";
    	std::cout << std::endl;
    
    	for (int i = 0; i < NUMBERS; i++)
    	    if (round_appearance[i] == 0 || (round_now - round_appearance[i] > k))
                std::cout << i << " ";
    	std::cout << std::endl;
            
    	if (input != 0) {
            if (input <= 10 ||(input > 18 && input <= 28))
                if (input % 2 == 1) ++reds;
                else                ++blacks;
            else
                if (input % 2 == 0) ++reds;
                else                ++blacks;
    	}
    	
        std::cout << reds << " " << blacks << std::endl;
        std::cout << std::endl;
    }
}
